spec/post_show_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'rails_helper'
+
+RSpec.describe 'PostShow', type: :system do
+  before do
+    FactoryBot.create(
+      :user,
+      name: 'Tom',
+      photo: 'https://picsum.photos/seed/1/200',
+      bio: 'An user with a bio.'
+    )
+
+    FactoryBot.create(
+      :post,
+      author: User.find_by(name: 'Tom'),
+      title: 'First Post by Tom',
+      text: 'This is the content of the first post by Tom.'
+    )
+
+    FactoryBot.create(
+      :comment,
+      post: Post.find_by(title: 'First Post by Tom'),
+      user: User.find_by(name: 'Tom'),
+      text: 'This is the first comment on the first post by Tom.'
+    )
+
+    FactoryBot.create(
+      :comment,
+      post: Post.find_by(title: 'First Post by Tom'),
+      user: User.find_by(name: 'Tom'),
+      text: 'This is the second comment on the first post by Tom.'
+    )
+
+    FactoryBot.create(
+      :comment,
+      post: Post.find_by(title: 'First Post by Tom'),
+      user: User.find_by(name: 'Tom'),
+      text: 'This is the third comment on the first post by Tom.'
+    )
+  end
+
+  it 'I can see the post title' do
+    tom = User.find_by(name: 'Tom')
+    post = Post.find_by(title: 'First Post by Tom')
+    visit user_post_path(tom, post)
+    expect(page).to have_content('First Post by Tom')
+  end
+
+  it 'I can see who wrote the post' do
+    tom = User.find_by(name: 'Tom')
+    post = Post.find_by(title: 'First Post by Tom')
+    visit user_post_path(tom, post)
+    expect(page).to have_content('Tom')
+  end
+
+  it 'I can see how many comments it has' do
+    tom = User.find_by(name: 'Tom')
+    post = Post.find_by(title: 'First Post by Tom')
+    visit user_post_path(tom, post)
+    expect(page).to have_content('Comments: 3')
+  end
+
+  it 'I can see how many likes it has' do
+    tom = User.find_by(name: 'Tom')
+    post = Post.find_by(title: 'First Post by Tom')
+    visit user_post_path(tom, post)
+    expect(page).to have_content('Likes: 0')
+  end
+
+  it 'I can see the post body' do
+    tom = User.find_by(name: 'Tom')
+    post = Post.find_by(title: 'First Post by Tom')
+    visit user_post_path(tom, post)
+    expect(page).to have_content('This is the content of the first post by Tom.')
+  end
+
+  it 'I can see the username of each commentor' do
+    tom = User.find_by(name: 'Tom')
+    post = Post.find_by(title: 'First Post by Tom')
+    visit user_post_path(tom, post)
+    expect(page).to have_content('Tom')
+  end
```

Comment:
Your test case for checking the commentor's username is on the right track, but there's a small issue. It appears that you're checking if the username 'Tom' is present on the page. However, this test case might not be very robust because it's specifically looking for 'Tom,' which is the post owner's username in this case.

To make the test case more valid, you should consider testing with a different username that is guaranteed to belong to a commenter, not the post owner. You can create a test scenario where a user with a different username leaves a comment, and then verify if that username is displayed on the page. This would ensure that the test checks for commentor usernames accurately.